src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,280 @@
+                    return await this.awaitProvisioningPipelineJob(jobId, githubOrg, repository, wizardInputs);
+                } else {
+                 throw new Error("Failed to receive queued pipeline provisioning job status");
+                }
+            }
+        } catch (error) {
+            throw error;
+        }
+    }
+
+    public async browseQueuedPipeline(): Promise<void> {
+        new ControlProvider().showInformationBox("Browse queued pipeline", Messages.githubWorkflowSetupSuccessfully, Messages.browseWorkflow)
+            .then((action: string) => {
+                if (action && action.toLowerCase() === Messages.browseWorkflow.toLowerCase()) {
+                    telemetryHelper.setTelemetry(TelemetryKeys.BrowsePipelineClicked, 'true');
+                    vscode.env.openExternal(vscode.Uri.parse(this.queuedPipelineUrl));
+                }
+            });
+    }
+
+    public async postSteps(provisioningConfiguration: ProvisioningConfiguration, draftPipelineConfiguration: DraftPipelineConfiguration, inputs: WizardInputs): Promise<void> {
+        await this.getFileToCommit(draftPipelineConfiguration);
+        await this.showPipelineFiles();
+        const displayMessage = (this.filesToCommit.length > 1) ?  Messages.modifyAndCommitMultipleFiles : Messages.modifyAndCommitFile;
+        const commitOrDiscard = await new ControlProvider().showInformationBox(
+            "Commit or discard",
+            utils.format(displayMessage, Messages.commitAndPush, inputs.sourceRepository.branch, inputs.sourceRepository.remoteName),
+            Messages.commitAndPush,
+            Messages.discardPipeline);
+        let provisioningServiceResponse: ProvisioningConfiguration;
+        if (!!commitOrDiscard && commitOrDiscard.toLowerCase() === Messages.commitAndPush.toLowerCase()) {
+             provisioningServiceResponse = await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification, title: Messages.configuringPipelineAndDeployment },
+             async () => {
+                try {
+                    telemetryHelper.setCurrentStep('QueuedCompleteProvisioiningPipeline');
+                    provisioningConfiguration.pipelineConfiguration = await this.createFilesToCheckin(draftPipelineConfiguration.id, draftPipelineConfiguration.type);
+                    const completeProvisioningSvcResp = await this.queueProvisioningPipelineJob(provisioningConfiguration, inputs);
+                    if ( completeProvisioningSvcResp.id != "" ){
+                        const OrgAndRepoDetails = inputs.sourceRepository.repositoryId.split('/');
+                        telemetryHelper.setCurrentStep('AwaitCompleteProvisioningPipeline');
+                        return await this.awaitProvisioningPipelineJob(completeProvisioningSvcResp.id, OrgAndRepoDetails[0], OrgAndRepoDetails[1], inputs);
+                    } else {
+                        throw new Error("Failed to configure pipeline");
```

Comment:
throw the error returned by API

Reply:
this is for same reason, response is 2xx. But payload doesn't have job id. So we are throwing our own error

Reply:
makes sense